WordPress/Classes/Services/PostCoordinator.swift

```diff
@@ -140,24 +140,29 @@ class PostCoordinator: NSObject {
     /// - warning: Work-in-progress (kahu-offline-mode)
     @MainActor
     func _publish(_ post: AbstractPost) async throws {
-        let post = post.original()
+        let original = post.original()
+        let latest = post.latest()
 
-        await pauseSyncing(for: post)
-        defer { resumeSyncing(for: post) }
+        await pauseSyncing(for: original)
+        defer { resumeSyncing(for: original) }
 
         var parameters = RemotePostUpdateParameters()
-        if post.status == .draft {
-            parameters.status = Post.Status.publish.rawValue
+        if original.status == .draft {
+            if let date = latest.dateCreated, date > .now {
+                parameters.status = Post.Status.scheduled.rawValue
+            } else {
+                parameters.status = Post.Status.publish.rawValue
```

Comment:
The `}  else { ` below will soon no longer be needed. The goal is to make sure `post.status` changes only after publishing.